receiptui/src/main/java/com/hyperwallet/android/ui/receipt/view/ListReceiptsFragment.java

```diff
@@ -0,0 +1,335 @@
+
+    @Override
+    public View onCreateView(@NonNull final LayoutInflater inflater, @Nullable final ViewGroup container,
+            @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_list_receipts, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull final View view, @Nullable final Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        mEmptyTransactionPlaceholder = view.findViewById(R.id.empty_transaction_list_view);
+        mListReceiptsView = view.findViewById(R.id.list_receipts);
+        mProgressBar = view.findViewById(R.id.list_receipt_progress_bar);
+        mListReceiptsAdapter = new ListReceiptsAdapter(mReceiptViewModel, new ListReceiptsItemDiffCallback());
+        mListReceiptsView.setHasFixedSize(true);
+        mListReceiptsView.setLayoutManager(new LinearLayoutManager(getActivity()));
+        mListReceiptsView.setAdapter(mListReceiptsAdapter);
+        registerObservers();
+    }
+
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        mReceiptViewModel.init();
+    }
+
+    @Override
+    public void onSaveInstanceState(@NonNull Bundle outState) {
+        outState.putBoolean(SHOULD_SHOW_NO_TRANSACTION_PLACEHOLDER, mShouldShowNoTransactionPlaceholder);
+        super.onSaveInstanceState(outState);
+        outState.putString(ARGUMENT_PREPAID_CARD_TOKEN, mPrepaidCardToken);
+    }
+
+    @Override
+    public void onViewStateRestored(@Nullable Bundle savedInstanceState) {
+        if (savedInstanceState != null) {
+            mShouldShowNoTransactionPlaceholder = savedInstanceState.getBoolean(SHOULD_SHOW_NO_TRANSACTION_PLACEHOLDER);
+        }
+        super.onViewStateRestored(savedInstanceState);
+        if (savedInstanceState != null) {
+            mPrepaidCardToken = savedInstanceState.getString(ARGUMENT_PREPAID_CARD_TOKEN);
+        } else {
+            //   mPrepaidCardToken = getArguments().getString(ARGUMENT_PREPAID_CARD_TOKEN);
+        }
+    }
+
+    private void registerObservers() {
+        mReceiptViewModel.receipts().observe(getViewLifecycleOwner(), new Observer<PagedList<Receipt>>() {
+            @Override
+            public void onChanged(final PagedList<Receipt> receipts) {
+                mListReceiptsAdapter.submitList(receipts);
+                receipts.addWeakCallback(null, new PagedList.Callback() {
+                    @Override
+                    public void onChanged(int position, int count) {
+                    }
+
+                    @Override
+                    public void onInserted(int position, int count) {
+                        if (receipts.size() > 0) {
+                            mShouldShowNoTransactionPlaceholder = Boolean.FALSE;
+                        } else {
+                            mShouldShowNoTransactionPlaceholder = Boolean.TRUE;
+                        }
```

Comment:
you can use ternary operator for such simple if/else

`mShouldShowNoTransactionPlaceholder = receipts.size() > 0 ? Boolean.FALSE : Boolean.TRUE;`